src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -25798,6 +25798,70 @@ public void Test1(IDo<object> impl)
             Assert.Equal("IResult<System.Int32>", doSymbol.ReturnType.ToTestDisplayString());
         }
 
+        [Fact]
+        [WorkItem(40430, "https://github.com/dotnet/roslyn/issues/40430")]
+        public void MissingTypeArgumentInBase_ValueTuple()
+        {
+            var lib_cs = @"
+public class ClassWithTwoTypeParameters<T1, T2>
+{
+}
+
+public class SelfReferencingClassWithTuple
+    : ClassWithTwoTypeParameters<SelfReferencingClassWithTuple, (string A, int B)>
+{
+}
+";
+            var lib = CreateCompilationWithMscorlib40(lib_cs, references: s_valueTupleRefs);
+            lib.VerifyDiagnostics();
+
+            var source_cs = @"
+public class TriggerStackOverflowException
+{
+    void Method()
+    {
+        _ = new SelfReferencingClassWithTuple();
+    }
+}
+";
+            var comp = CreateCompilationWithMscorlib40(source_cs, references: new[] { lib.EmitToImageReference() });
```

Comment:
>lib.EmitToImageReference() [](start = 86, length = 26)

Should test with compilation reference as well.

Reply:
Sure, I'll add. 
From my investigation on this issue, compilation references did not repro the overflow.

---
In reply to: [362910356](https://github.com/dotnet/roslyn/pull/40704#discussion_r362910356) [](ancestors = 362910356)

Reply:
> From my investigation on this issue, compilation references did not repro the overflow.

That doesn't surprise me, because the attribute encoding/decoding wouldn't be involved. However, we want consistent compiler's behavior across source and metadata, especially in such cases. Having a test allows us to verify and enforce that. 

---
In reply to: [362927964](https://github.com/dotnet/roslyn/pull/40704#discussion_r362927964) [](ancestors = 362927964,362910356)